source/schwarz_base.cpp

```diff
@@ -339,10 +365,31 @@ void SchwarzBase<ValueType, IndexType>::run(
     auto start_time = std::chrono::steady_clock::now();
     int num_converged_procs = 0;
 
+    //CHANGED - file writing
+    char send_name[30], recv_name[30], pe_str[3];
+    sprintf(pe_str, "%d", metadata.my_rank);
```

Comment:
Can you move these file writings to a function and out of the timed section.

Reply:
I need to print this during the solver to see when the events are being triggered. I opened and closed the file outside the loop. Then I checked that file writing and no file writing during the loop did not have noticeable difference in timing. Can you think of a better way to do it?

Reply:
You can store it in a variable and then print it later. file I/O is not a good idea within any timed loop.